src/msg/async/AsyncConnection.cc

```diff
@@ -868,7 +868,7 @@ ssize_t AsyncConnection::_process_connection()
         if (r < 0)
           goto fail;
 
-        center->create_file_event(cs.fd(), EVENT_READABLE, read_handler);
+        center->create_file_event(cs.fd(), EVENT_WRITABLE, read_handler);
```

Comment:
On Sat, Sep 3, 2016 at 1:21 PM, zpllieo <notifications@github.com> wrote:

> In src/msg/async/AsyncConnection.cc
> <https://github.com/ceph/ceph/pull/10968#discussion_r77431671>:
>
> > @@ -868,7 +868,7 @@ ssize_t AsyncConnection::_process_connection()
> >          if (r < 0)
> >            goto fail;
> >
> > -        center->create_file_event(cs.fd(), EVENT_READABLE, read_handler);
> > +        center->create_file_event(cs.fd(), EVENT_WRITABLE, read_handler);
>
> I got it,
> On the one hand, whether we can through macro definition to solve the
> problem, on the other hand, is it possible to fd has Always can't readable?
> use "center->dispatch_event_external(read_handler)" is better at this
> moment?
>

macro definition?

Like mentioned before, I think for general purpose EVENT_READABLE is
suitable for is_connect semantic. But posix require EVENT_WRITABLE, so I
think in current async networkstack abstract define,
EVENT_READABLE|EVENT_WRITABLE is choose by backend  which is used to notify
connection built.


> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ceph/ceph/pull/10968/files/7490fb519ec54278c711f899b9a523029af05de4#r77431671>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA72YZokUnmN-skNs7L0Zydm1oQ2LcCSks5qmQPlgaJpZM4J0N9O>
> .
>



-- 

Best Regards,

Wheat